saos-restful-web-api/src/test/java/pl/edu/icm/saos/api/dump/enrichmenttag/DumpEnrichmentTagControllerTest.java

```diff
@@ -0,0 +1,199 @@
+            .andExpect(jsonPath("$.items.[0].tagType").value("REFERENCED_REGULATIONS"))
+            .andExpect(jsonPath("$.items.[0].value.ref").value("AAA1"))
+        
+            .andExpect(jsonPath("$.items.[1].id").value(enrichmentTags.get(1).getId()))
+            .andExpect(jsonPath("$.items.[1].judgmentId").value(enrichmentTags.get(1).getJudgmentId()))
+            .andExpect(jsonPath("$.items.[1].tagType").value("REFERENCED_CASE_NUMBERS"))
+            .andExpect(jsonPath("$.items.[1].value.caseNumbers.[0]").value("XYZ1"))
+            .andExpect(jsonPath("$.items.[1].value.caseNumbers.[1]").value("XYZ2"))
+        
+            .andExpect(jsonPath("$.items.[2].id").value(enrichmentTags.get(2).getId()))
+            .andExpect(jsonPath("$.items.[2].judgmentId").value(enrichmentTags.get(2).getJudgmentId()))
+            .andExpect(jsonPath("$.items.[2].tagType").value("KEYWORDS"))
+            .andExpect(jsonPath("$.items.[2].value.keywords.[0]").value("val11"))
+            .andExpect(jsonPath("$.items.[2].value.keywords.[1]").value("val21"))
+            .andExpect(jsonPath("$.items.[2].value.keywords.[2]").value("val32"));
+        
+    }
+    
+    @Test
+    public void showEnrichmentTags_PAGING() throws Exception {
+        
+        // given
+        
+        int pageSize = 2;
+        int pageNumber = 1;
+        
+        
+        // execute
+        
+        ResultActions result = mockMvc.perform(get(DUMP_ENRICHMENT_PATH)
+                .param(PAGE_SIZE, String.valueOf(pageSize))
+                .param(PAGE_NUMBER, String.valueOf(pageNumber))
+                .accept(MediaType.APPLICATION_JSON));
+        
+        
+        // assert
+        
+        result
+            .andExpect(jsonPath("$.items", hasSize(1)))
+            
+            .andExpect(jsonPath("$.items.[0].id").value(enrichmentTags.get(2).getId()))
+            .andExpect(jsonPath("$.items.[0].judgmentId").value(enrichmentTags.get(2).getJudgmentId()))
+            .andExpect(jsonPath("$.items.[0].tagType").value("KEYWORDS"))
+            .andExpect(jsonPath("$.items.[0].value.keywords.[0]").value("val11"))
+            .andExpect(jsonPath("$.items.[0].value.keywords.[1]").value("val21"))
+            .andExpect(jsonPath("$.items.[0].value.keywords.[2]").value("val32"))
+
+            .andExpect(jsonPath("$.queryTemplate.pageSize.value").value(pageSize))
+            .andExpect(jsonPath("$.queryTemplate.pageNumber.value").value(pageNumber));
+    }
+    
+    @Test
+    public void showEnrichmentTags_WRONG_PARAM() throws Exception {
+        
+        // execute
+        
+        ResultActions actions = mockMvc.perform(get(DUMP_ENRICHMENT_PATH)
+                .param("some_incorrect_parameter_name", "")
+                .accept(MediaType.APPLICATION_JSON));
+
+        
+        // assert
+        
+        actions.andExpect(status().isBadRequest());
```

Comment:
We should check here if the client gets a message indicating the wrong param. The standard response is like this:
    {

    "error": 

    {
        "httpStatus": "400",
        "name": "WRONG REQUEST PARAMETER",
        "propertyName": "pageNu",
        "message": "parameter 'pageNu' : name is incorrect",
        "moreInfo": "${restful.api.error.documentation.site}WRONG_REQUEST_PARAMETER_ERROR"
       }

    }